pkg/liqo-controller-manager/internal-network/internalfabric-controller/internalfabric_controller.go

```diff
@@ -65,10 +70,89 @@ func (r *InternalFabricReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	if !internalFabric.DeletionTimestamp.IsZero() && controllerutil.ContainsFinalizer(internalFabric, consts.InternalFabricGeneveTunnelFinalizer) {
+		// delete geneve tunnels
+		var geneveTunnelList networkingv1alpha1.GeneveTunnelList
+		if err = r.List(ctx, &geneveTunnelList, client.MatchingLabels{
+			consts.InternalFabricName:      internalFabric.Name,
+			consts.InternalFabricNamespace: internalFabric.Namespace,
```

Comment:
Is the namespace label necessary in this case? We could list the tunnels in the fabric namespace using the `client.InNamespace()` option.